fix(formatter): should not add a hard space before function body

Dunqing opened the pull request:
This is a regression I found in the VSCode repo, caused by #16093. This is a mistake I made, as we should never use a hard space.

leaysgur commented:
### Merge activity

* **Nov 28, 2:42 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 28, 2:42 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(formatter): should not add a hard space before function body (#16221)

This is a regression I found in the VSCode repo, caused by #16093. This is a mistake I made, as we should never use a hard space.

leaysgur commented on crates/oxc_formatter/tests/fixtures/js/arrow-function/object-method.ts:
Nits, no problem! 😉
